path_login.go

```diff
@@ -273,8 +274,33 @@ func (b *azureAuthBackend) verifyResource(ctx context.Context, subscriptionID, r
 			principalIDs[to.String(vmss.Identity.PrincipalID)] = struct{}{}
 		}
 		// if not, look for user-assigned identities
-		for _, userIdentity := range vmss.Identity.UserAssignedIdentities {
-			principalIDs[to.String(userIdentity.PrincipalID)] = struct{}{}
+		for userIdentityID, userIdentity := range vmss.Identity.UserAssignedIdentities {
+			// Principal ID is not nil for VMSS uniform orchestration mode
+			if userIdentity.PrincipalID != nil {
+				principalIDs[to.String(userIdentity.PrincipalID)] = struct{}{}
+				continue
+			}
+
+			elements := strings.Split(userIdentityID, "/")
+			if len(elements) < 9 {
+				return fmt.Errorf("unable to parse the userIdentityID: %s", userIdentityID)
+			}
+			msiSubscriptionID := elements[2]
+			msiResourceGroupName := elements[4]
+			msiResourceName := elements[8]
+
```

Comment:
```suggestion
	// Principal ID is nil for VMSS flex orchestration mode, so we 
	// must look up the user-assigned identity using the MSI client
```

Reply:
Suggest to add this comment to explain the non-obvious bits of why we're doing this for future readers.

Reply:
Sure